### ISSUE TITLE
`wallet.sync()` is scanning 200 addresses

### ISSUE THREAD

**Issue by Czino** [LtbLightning/bdk-rn#44]
When syncing, even after after creating a fresh wallet with a `stopGap` setting of `5`, I am logging 200 requests on my electrum server.

Changing the `stopGap` does not have any influence on these 200 requests.

I cannot find any setting to change the number of initial addresses to be scanned but I would expect it to be much lower (10/25/50)

**Comment by Czino** [comment]
So far debugging and finding that `stopGap` and `timeout` and mixed up for Esplora

in `src/classes/Blockchain.ts:31`
stopGap is the 5th argument
`this.id = await this._bdk.initEsploraBlockchain(url, proxy, concurrency, timeout, stopGap);`

`android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt:232`
```
@ReactMethod
    fun initEsploraBlockchain(
        url: String,
        proxy: String?,
        concurrency: String?,
        stopGap: String?,
        timeOut: String?,
        result: Promise
    )
```

**Comment by Czino** [comment]
After fixing it with https://github.com/LtbLightning/bdk-rn/pull/46, I inspected the `stopGap` in android and now I receive the same value I passed.

However, bdk-rn is still scanning 200 addresses.

**Comment by BitcoinZavior** [comment]
Thanks, @Czino will debug and see what could be wrong.

**Comment by BitcoinZavior** [comment]
@Czino you might also want to check the value for `concurrency: String? `
https://docs.rs/bdk/latest/bdk/blockchain/esplora/struct.EsploraBlockchainConfig.html

**Comment by Czino** [comment]
@BitcoinZavior yes, I am already using `concurrency` to speed things up a little

**Comment by Czino** [comment]
debugging this issue a little, even after forcing `stopGap` to be `1`, still 200 addresses are being scanned

```
 let _blockchainConfig = BlockchainConfig.esplora(
    config: EsploraConfig(
        baseUrl: baseUrl,
        proxy: proxy.isEmpty ? nil : proxy,
        concurrency: UInt8(truncating: concurrency),
        stopGap: 1,
        timeout: UInt64(truncating: timeout)
    )
)
```

Looks to me like we need to escalate the issue to bdk-ffi
